src/context.cc

```diff
@@ -263,30 +244,44 @@ void ContextBase::onForeignFunction(uint32_t foreign_function_id, uint32_t data_
 }
 
 FilterStatus ContextBase::onNetworkNewConnection() {
-  CHECK_NET(on_new_connection_, FilterStatus::Continue, FilterStatus::StopIteration);
-  DeferAfterCallActions actions(this);
-  if (wasm_->on_new_connection_(this, id_).u64_ == 0) {
+  CHECK_FAIL_NET(FilterStatus::Continue, FilterStatus::StopIteration);
+  if (!wasm_->on_new_connection_) {
     return FilterStatus::Continue;
   }
-  return FilterStatus::StopIteration;
+  DeferAfterCallActions actions(this);
+  const auto result = wasm_->on_new_connection_(this, id_).u64_;
+  CHECK_FAIL_NET(FilterStatus::Continue, FilterStatus::StopIteration);
+  return result == 0 ? FilterStatus::Continue : FilterStatus::StopIteration;
 }
 
 FilterStatus ContextBase::onDownstreamData(uint32_t data_length, bool end_of_stream) {
-  CHECK_NET(on_downstream_data_, FilterStatus::Continue, FilterStatus::StopIteration);
+  CHECK_FAIL_NET(FilterStatus::Continue, FilterStatus::StopIteration);
+  if (!wasm_->on_downstream_data_) {
+    return FilterStatus::Continue;
+  }
   DeferAfterCallActions actions(this);
-  auto result = wasm_->on_downstream_data_(this, id_, static_cast<uint32_t>(data_length),
-                                           static_cast<uint32_t>(end_of_stream));
+  auto result = wasm_
+                    ->on_downstream_data_(this, id_, static_cast<uint32_t>(data_length),
+                                          static_cast<uint32_t>(end_of_stream))
+                    .u64_;
   // TODO(PiotrSikora): pull Proxy-WASM's FilterStatus values.
-  return result.u64_ == 0 ? FilterStatus::Continue : FilterStatus::StopIteration;
+  CHECK_FAIL_NET(FilterStatus::Continue, FilterStatus::StopIteration);
+  return result == 0 ? FilterStatus::Continue : FilterStatus::StopIteration;
 }
 
 FilterStatus ContextBase::onUpstreamData(uint32_t data_length, bool end_of_stream) {
-  CHECK_NET(on_upstream_data_, FilterStatus::Continue, FilterStatus::StopIteration);
+  CHECK_FAIL_NET(FilterStatus::Continue, FilterStatus::StopIteration);
+  if (!wasm_->on_upstream_data_) {
+    return FilterStatus::Continue;
+  }
   DeferAfterCallActions actions(this);
-  auto result = wasm_->on_upstream_data_(this, id_, static_cast<uint32_t>(data_length),
-                                         static_cast<uint32_t>(end_of_stream));
+  auto result = wasm_
+                    ->on_upstream_data_(this, id_, static_cast<uint32_t>(data_length),
+                                        static_cast<uint32_t>(end_of_stream))
+                    .u64_;
```

Comment:
Nit: This is consistent with rest of the file, but I don't think we should be unpacking `Word` and accessing `u64_` to store the call result.